cmd/kubelet/app/server.go

```diff
@@ -418,11 +425,13 @@ func InitializeTLS(s *options.KubeletServer) (*server.TLSOptions, error) {
 	if s.TLSCertFile == "" && s.TLSPrivateKeyFile == "" {
 		s.TLSCertFile = path.Join(s.CertDirectory, "kubelet.crt")
 		s.TLSPrivateKeyFile = path.Join(s.CertDirectory, "kubelet.key")
-		if crypto.ShouldGenSelfSignedCerts(s.TLSCertFile, s.TLSPrivateKeyFile) {
-			if err := crypto.GenerateSelfSignedCert(nodeutil.GetHostname(s.HostnameOverride), s.TLSCertFile, s.TLSPrivateKeyFile, nil, nil); err != nil {
-				return nil, fmt.Errorf("unable to generate self signed cert: %v", err)
+
+		// TODO(yifan): We should also make sure that certificate contains
+		// accurate Common name and SANs for the current hostname and IP.
+		if !crypto.FoundCertOrKey(s.TLSCertFile, s.TLSPrivateKeyFile) {
+			if err := createCertAndKey(s); err != nil {
```

Comment:
We need to use this for the client cert (to configure the RESTClient used to connect to the apiserver). We should not use this for the serving cert.